### PR TITLE
ELECTRON-357 (Fix time-out issue in Unit tests and Spectron tests.)

### PR DESCRIPTION
**Pull request by keerthi16**
## Description
Fixes time-out issue in Unit tests and Spectron tests.
Changed the unit tests time-out to 60000s.

**Before:**
150s to 200s to run the Unit tests
300s to 400s to run the Spectron tests

**After:** 
30s to 35s to run the Unit tests
80s to 90s to run the Spectron tests

## Approach
How does this change address the problem?
- #### Problem with the code: Running `rebuild` script every time. causing time-out 
- #### Fix: Now running the `rebuild`script once before starting the tests.

## Learning
N/A

#### Blog Posts
N/A

## Related PRs
N/A

## Open Questions if any and Todos
- [x] Unit-Tests
Build number: #30
screen shot 2018-03-13 at 11 01 03 am

Spectron
screen shot 2018-03-13 at 12 53 17 pm

- [x] Documentation
- [x] Automation-Tests

@VikasShashidhar @VishwasShashidhar @KiranNiranjan  Please review. Thanks 😃

### CODE REVIEW

File: tests/spectron/spellChecker.spectron.js [hunk 1]
```diff
@@ -13,16 +12,14 @@ describe('Tests for spellChecker', () => {
...
-            robot = require('robotjs');
-            app.startApplication().then((startedApp) => {
-                app = startedApp;
-                done();
-            }).catch((err) => {
-                console.error(`Unable to start application error: ${err}`);
-                expect(err).toBeNull();
-                done();
-            });
+        robot = require('robotjs');
```
Comment (KiranNiranjan) [inline tests/spectron/spellChecker.spectron.js]: You can require `robotjs` globally now 😄

Comment (keerthi16) [reply tests/spectron/spellChecker.spectron.js]: Done 😄

File: tests/activityDetection.test.js [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
 
     beforeAll(function(done) {
-        childProcess.exec(`npm rebuild --target=${process.version} --build-from-source`, function(err) {
-            activityDetection = require('../js/activityDetection');
-            activityDetection.setActivityWindow(900000, electron.ipcRenderer);
-            done();
-        });
+        activityDetection.setActivityWindow(900000, electron.ipcRenderer);
```
Comment (VishwasShashidhar) [inline tests/activityDetection.test.js]: is 90 seconds required for this? if not, can you use the above variable?

File: tests/spectron/close.spectron.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ let app = new Application({});
 describe('Tests for Close', () => {
 
     let originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = Application.getTimeOut();
```
Comment (VishwasShashidhar) [inline tests/spectron/close.spectron.js]: what `Application.getTimeOut()` does exactly? and, where does it get the timeout information from?

File: tests/spectron/full-screen.spectron.js [hunk 4]
```diff
@@ -1,35 +1,31 @@
...
 
-let robot;
 let configPath;
 let app = new Application({});
 
 describe('Tests for Full screen', () => {
 
     let originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = Application.getTimeOut();
```
Comment (VishwasShashidhar) [inline tests/spectron/full-screen.spectron.js]: what `Application.getTimeOut()` does exactly? and, where does it get the timeout information from?

File: tests/spectron/minimize-on-close.spectron.js [hunk 5]
```diff
@@ -1,35 +1,31 @@
...
 
-let robot;
 let configPath;
 let app = new Application({});
 
 describe('Tests for Minimize on Close', () => {
 
     let originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = Application.getTimeOut();
```
Comment (VishwasShashidhar) [inline tests/spectron/minimize-on-close.spectron.js]: what `Application.getTimeOut()` does exactly? and, where does it get the timeout information from?

File: tests/spectron/zoom-in-zoom-out.spectron.js [hunk 6]
```diff
@@ -1,36 +1,32 @@
...
-let robot;
 let configPath;
 
 let app = new Application({});
 
 describe('Tests for Zoom in and Zoom out', () => {
 
     let originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = Application.getTimeOut();
```
Comment (VishwasShashidhar) [inline tests/spectron/zoom-in-zoom-out.spectron.js]: what `Application.getTimeOut()` does exactly? and, where does it get the timeout information from?

File: tests/spectron/spellChecker.spectron.js [hunk 7]
```diff
@@ -1,28 +1,23 @@
...
-                console.error(`Unable to start application error: ${err}`);
-                expect(err).toBeNull();
-                done();
-            });
+        app.startApplication().then((startedApp) => {
+            app = startedApp;
+            done();
+        }).catch((err) => {
+            console.error(`Unable to start application error: ${err}`);
+            expect(err).toBeNull();
```
Comment (VishwasShashidhar) [inline tests/spectron/spellChecker.spectron.js]: why this? when we are in the catch section, we have to expect err to be not null right?

File: tests/spectron/zoom-in-zoom-out.spectron.js [hunk 8]
```diff
@@ -1,36 +1,32 @@
...
-                });
+        return app.startApplication().then((startedApp) => {
+            app = startedApp;
+            getConfigPath().then((config) => {
+                configPath = config;
+                done();
             }).catch((err) => {
-                console.error(`Unable to start application error: ${err}`);
+                console.error(`Unable to get user config path error: ${err}`);
                 expect(err).toBeNull();
```
Comment (VishwasShashidhar) [inline tests/spectron/zoom-in-zoom-out.spectron.js]: why this? when we are in the catch section, we have to expect err to be not null right?

File: tests/spectron/zoom-in-zoom-out.spectron.js [hunk 9]
```diff
@@ -1,36 +1,32 @@
...
                 done();
             });
+        }).catch((err) => {
+            console.error(`Unable to start application error: ${err}`);
+            expect(err).toBeNull();
```
Comment (VishwasShashidhar) [inline tests/spectron/zoom-in-zoom-out.spectron.js]: why this? when we are in the catch section, we have to expect err to be not null right?

**Comment by VishwasShashidhar** [comment]
@keerthi16 Please address the changes I've requested as part of the comments.

Also, squash the commits and add title to the commits along with the ticket number.

Thanks!

File: tests/spectron/spellChecker.spectron.js (hunk 7 above)
Comment (keerthi16) [reply tests/spectron/spellChecker.spectron.js]: Yes, we are expecting the error to be null so that the test should fail.

**Comment by VishwasShashidhar** [comment]
lgtm 👍

**Pull request merged**